custom/icds_reports/management/commands/national_user_data_pull.py

```diff
@@ -0,0 +1,45 @@
+import csv
+
+from django.core.management.base import BaseCommand
+from corehq.apps.users.models import CommCareUser
+from custom.icds_reports.const import INDIA_TIMEZONE
+from custom.icds_reports.models import ICDSAuditEntryRecord
+from django.db.models import Max
+from dimagi.utils.chunked import chunked
+
+
+class Command(BaseCommand):
+    help = "Custom data pull"
+
+    def convert_to_ist(self, date):
+        if date is None:
+            return 'N/A'
+        date = date.astimezone(INDIA_TIMEZONE)
+        date_formatted = date.strftime("%d/%m/%Y, %I:%M %p")
+        return date_formatted
+
+    def handle(self, *args, **options):
+        users = CommCareUser.by_domain('icds-cas')
+        usernames = []
+        for user in users:
+            if user.has_permission('icds-cas', 'access_all_locations'):
```

Comment:
it turns out that finding national level is trickier. I think a better condition would be:
```
if ((len(user.assigned_location_ids)==0 or 'cpmu' in user.username.tolower() ) and user.has_permission('icds-cas', 'access_all_locations')):
```